tests/bsim/bluetooth/mesh/src/test_beacon.c

```diff
@@ -1695,6 +1695,68 @@ static void test_rx_priv_multi_net_id(void)
 
 	PASS();
 }
+
+static void test_tx_priv_gatt_proxy(void)
+{
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+	bt_mesh_device_setup(&prov, &prb_comp);
+	provision(&rx_cfg);
+	bt_mesh_iv_update_test(true);
+
+	ASSERT_TRUE(bt_mesh.iv_index == 0);
+
+	/* Disable SNB. */
+	bt_mesh_beacon_set(false);
+	ASSERT_OK_MSG(bt_mesh_scan_disable(), "Failed to disable scanner");
+	ASSERT_OK_MSG(bt_mesh_gatt_proxy_set(BT_MESH_GATT_PROXY_DISABLED),
+		      "Failed to disable gatt proxy");
+	ASSERT_OK_MSG(bt_mesh_priv_gatt_proxy_set(BT_MESH_PRIV_GATT_PROXY_ENABLED),
+		      "Failed to set private gatt proxy");
+
+	/* Wait for proxy connection to complete. */
+	k_sleep(K_SECONDS(10));
+
+	/* Start IV update */
+	ASSERT_TRUE(bt_mesh_iv_update());
+
+	/* Check that IV index has updated */
+	ASSERT_TRUE(bt_mesh.iv_index == 1);
+	PASS();
+}
+
+static void test_rx_priv_gatt_proxy(void)
+{
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+	bt_mesh_device_setup(&prov, &prb_comp);
+	provision(&tx_cfg);
+	bt_mesh_iv_update_test(true);
+
+	ASSERT_TRUE(bt_mesh.iv_index == 0);
+
+	/* Disable SNB. */
+	bt_mesh_beacon_set(false);
+	ASSERT_OK_MSG(bt_mesh_gatt_proxy_set(BT_MESH_GATT_PROXY_DISABLED),
+		      "Failed to disable gatt proxy");
+	ASSERT_OK_MSG(bt_mesh_priv_gatt_proxy_set(BT_MESH_PRIV_GATT_PROXY_ENABLED),
+		      "Failed to set private gatt proxy");
+	ASSERT_OK_MSG(bt_mesh_proxy_connect(TEST_NET_IDX1), "Failed to connect over proxy");
+
+	/* Wait for connection to complete, then disable scanner
+	 * to ensure that all RX communication arrives over GATT.
+	 */
+	k_sleep(K_SECONDS(5));
```

Comment:
How do you ensure that establishing a connection doesn't take 5 or more seconds so that the 10 seconds sleep expires on the other device before this one disables scanner?

Reply:
Trial and error. I have verified from the logs that it is sufficient but I would prefer something a little less fluffy. Like a `connected` cb, "conn_get" API or something, but I did not find any. I'm open for suggestions.

Reply:
I mean, in future, if something changes, how can we be still sure that the beacon is sent after scanner is disabled?

Reply:
Well we can not 100% guarantee that with sleeps. As far as I know the sync functionality is not available anymore in Bsim and there is no API for cheking if the connection is established. So then my best suggestion is some ad hoc message sending to create sync between tx and rx, but it is far from optimal, or add API that can get the status of a connection.

Reply:
What if you disable the scanner right after provisioning?

Reply:
I think we need the scanner active until the connection is in place so that it can receive priv net id beacons. And also, `bt_mesh_proxy_connect` enables the scanner after connection completes unconditionally.

Reply:
Ok. My initial though was to implement the connection right in the test. Then we can be certain that we receive the beacon over GATT. This doesn't need an extra logic before establishing the connection that Proxy Client has. I don't have other ideas. If you think the current test is enough, then I will approve this PR.

Reply:
>My initial though was to implement the connection right in the test. Then we can be certain that we receive the beacon over GATT. This doesn't need an extra logic before establishing the connection that Proxy Client has.

I was considering this at first but the test adds more coverage if we include both SRV and CLI side.

Reply:
> ```
> bool bt_mesh_proxy_cli_is_connected(uint16_t net_idx)
> {
> 	if (find_proxy_srv(net_idx, false, true)) {
> 		return true;
> 	}
> 
> 	return false;
> }
> ```
> 
> How about adding this as part of the internal API? For both SRV and CLI side. At some point we will need to do more testing on GATT proxy, so having some utility API would make that a lot easier.

This may work!

> > My initial though was to implement the connection right in the test. Then we can be certain that we receive the beacon over GATT. This doesn't need an extra logic before establishing the connection that Proxy Client has.
> 
> I was considering this at first but the test adds more coverage if we include both SRV and CLI side.

It is not good to test an implementation against itself as if there is the same bug in both sides it may not be caught.

Reply:
>It is not good to test an implementation against itself as if there is the same bug in both sides it may not be caught.

Yes, you got a strong point there. Let me do my suggested changes and lets review it again after that.

Reply:
We test everything against itself. adv against scanner, transport tx against transport rx, provisioner against provisionee etc etc etc. Why should create the separate gatt proxy client for testing? This will cause unnecessary bsim test maintenance any time when we add something into gatt proxy server.